web/app.js

```diff
@@ -687,11 +686,11 @@ var PDFViewerApplication = {
     var loadingTask = pdfjsLib.getDocument(parameters);
     this.pdfLoadingTask = loadingTask;
 
-    loadingTask.onPassword = function passwordNeeded(updatePassword, reason) {
-      PasswordPrompt.updatePassword = updatePassword;
-      PasswordPrompt.reason = reason;
-      PasswordPrompt.open();
-    };
+    loadingTask.onPassword = function passwordNeeded(callback, reason) {
+      this.passwordPrompt.setCallback(callback);
+      this.passwordPrompt.setReason(reason);
```

Comment:
Well, in e.g. [`PDFLinkService.setDocument`](https://github.com/mozilla/pdf.js/blob/master/web/pdf_link_service.js#L51), we're passing in more than one argument.
So I suppose that e.g. `self.passwordPrompt.setCallback(updateCallback, reason)` could work?
Or if you want do as in [`PDFDocumentProperties.setDocumentAndUrl`](https://github.com/mozilla/pdf.js/blob/master/web/pdf_document_properties.js#L109), you could do e.g. `self.passwordPrompt.setCallbackAndReason(updateCallback, reason)` instead?